### PR TITLE
[2.x] Split up quantity select into 2 variants

### PR DESCRIPTION
**Pull request by BobWez98**
Once this is merged i'll add the PR for v3.

This PR splits up the quantity select in 2 variants; a default and a enhanced version with plus and minus buttons. They use the same props so they can be switched seamlessly. 
The use on the product page: 
```
<x-rapidez::quantity :minSaleQty="$product->min_sale_qty" model="addToCart.qty" :qtyIncrements="$product->qty_increments" />
```
Or
```
<x-rapidez::quantity.enhanced :minSaleQty="$product->min_sale_qty" model="addToCart.qty" :qtyIncrements="$product->qty_increments" />
```

This component can also be used in the cart, we replace the form [here](https://github.com/rapidez/core/blob/master/resources/views/cart/item/quantity.blade.php#L8) by:
```
<div class="flex items-center gap-1">
        <x-rapidez::quantity @change="mutate" index="item.id" model="variables.quantity" minSaleQty="(item.product.stock_item?.min_sale_qty ?? 1)" qtyIncrements="(item.product.stock_item?.qty_increments ?? 1)"/>
</div>
```
The only difference is that the default quantity needs an extra `@change` eventlistener.
or 
```
<div class="flex items-center gap-1">
        <x-rapidez::quantity.enhanced index="item.id" model="variables.quantity" minSaleQty="(item.product.stock_item?.min_sale_qty ?? 1)" qtyIncrements="(item.product.stock_item?.qty_increments ?? 1)"/>
</div>
```

Here we use the index to determine which input to update. In the EnhancedQuantitySelect.vue.

### CODE REVIEW

File: resources/js/components/GraphqlMutation.vue [hunk 1]
```diff
@@ -93,7 +93,7 @@ export default {
 
     mounted() {
         if (this.mutateEvent) {
-            window.app.$on(this.mutateEvent, () => {
+            this.$root.$on(this.mutateEvent, () => {
```
Comment (BobWez98) [inline resources/js/components/GraphqlMutation.vue:96]: In the cart this caused an error; `window.app` is not defined

**Comment by BobWez98** [comment]
Frontend will be done by @Roene-JustBetter

**Review by Roene-JustBetter — DISMISSED** [review]

File: resources/js/components/Product/EnhancedQuantitySelect.vue [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        },
+
+        decrease() {
+            if (this.model - this.increment >= this.minQty) {
+                this.updateQty(this.model - this.increment)
+            }
+        },
+
+        updateQty(qty) {
+            let input = this.$scopedSlots.default()[0].context.$refs[this.inputRef]
```
Comment (royduin) [inline resources/js/components/Product/EnhancedQuantitySelect.vue:42]: Can't we use the model?

**Review by royduin — CHANGES_REQUESTED** [review]

File: resources/js/components/Product/EnhancedQuantitySelect.vue (hunk 2 above)
Comment (BobWez98) [reply resources/js/components/Product/EnhancedQuantitySelect.vue:42]: No, i did that before, however this won't work when in the cart. First i had a event listener on the `addtocart.vue` that would change the model qty. This won't work with the graphql-mutation component that handles it in the cart. This works for all scenarios

File: resources/views/components/quantity/index.blade.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+<x-rapidez::select
+    {{ $attributes->class('w-auto') }}
+    name="qty"
+    label="Quantity"
+    v-model="{{ $model }}"
+    labelClass="flex items-center sr-only mr-3"
+    wrapperClass="flex"
+>
+    <option v-for="i in ({{ $qtyIncrements }} * 10)" v-if="i % {{ $qtyIncrements }} === 0" v-bind:value="i" v-text="i"></option>
```
[the remark this answers is not in the document]
Comment (BobWez98) [reply resources/views/components/quantity/index.blade.php:15]: We need the v-if for occurances of the `qtyIncrements` not being 1. 
If the `qtyIncrements` is 10, we don't want 1,2,3,4,5 etc in the dropdown.

**Comment by BobWez98** [comment]
Feedback is processed.

**Review by Roene-JustBetter — DISMISSED** [review]

**Review by indykoning — DISMISSED** [review]

**Comment by BobWez98** [comment]
V3: https://github.com/rapidez/core/pull/608

**Review by indykoning — APPROVED** [review]

**Comment by royduin** [comment]
Closing this in favor of https://github.com/rapidez/core/pull/645

**Pull request closed without merging**